tardis/io/model/readers/generic_readers.py

```diff
@@ -0,0 +1,403 @@
+from astropy import units as u
+from numpy import recfromtxt
+import pandas as pd
+from radioactivedecay import Nuclide
+from radioactivedecay.utils import Z_DICT, elem_to_Z
+
+
+import numpy as np
+from tardis.io.model.readers.cmfgen import (
+    read_cmfgen_composition,
+    read_cmfgen_density,
+)
+
+from tardis.util.base import parse_quantity
+
+class ConfigurationError(Exception):
+    pass
+
+
+
+def read_simple_ascii_density(fname: str):
+    """
+    Reading a density file of the following structure (example; lines starting with a hash will be ignored):
+    The first density describes the mean density in the center of the model and is not used.
+    5 s
+    #index velocity [km/s] density [g/cm^3]
+    0 1.1e4 1.6e8
+    1 1.2e4 1.7e8
+
+    Parameters
+    ----------
+    fname : str
+        filename or path with filename
+
+    Returns
+    -------
+    time_of_model : astropy.units.Quantity
+        time at which the model is valid
+    data : pandas.DataFrame
+        data frame containing index, velocity (in km/s) and density
+    """
+
+    with open(fname) as fh:
+        time_of_model_string = fh.readline().strip()
+        time_of_model = parse_quantity(time_of_model_string)
+
+    data = recfromtxt(
+        fname,
+        skip_header=1,
+        names=("index", "velocity", "density"),
+        dtype=(int, float, float),
+    )
+    velocity = (data["velocity"] * u.km / u.s).to("cm/s")
+    mean_density = (data["density"] * u.Unit("g/cm^3"))[1:]
+
+    return time_of_model, velocity, mean_density
+
+
+def read_artis_density(fname):
```

Comment:
Is there a reason the artis reader lives in the generic readers file? Would it be more clear to live in an artist reader file similar to cmfgen currently?

Reply:
Agreed

Reply:
that should have been removed and moved to `artis.py`. It's not showing up in the diff but is definitely there. check it.